packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -151,10 +151,7 @@ export default function createCssVarsProvider(options) {
     const defaultColorSchemeStyleSheet = {};
     const otherColorSchemesStyleSheet = {};
     Object.entries(colorSchemes).forEach(([key, scheme]) => {
-      const { css, vars } = cssVarsParser(scheme, {
-        prefix: cssVarPrefix,
-        shouldSkipGeneratingVar,
-      });
+      const { css, vars } = generateCssVars(key);
       theme.vars = deepmerge(theme.vars, vars);
```

Comment:
One last comment, should we remove the merging here as well? Since `vars` comes from the `theme` input, there is no need to do it all over again.

If the developer wants to modify the `theme.vars`, the `resolveTheme()` is the escape hatch.

From a performance perspective, I log the time used to calculate the theme in `CssVarsProvider`:

**With merging**: ~0.3ms
**Without merging**: ~0.1ms

Even though it does not differ much but if the component rerenders multiple times, it could be slightly different.